### PR TITLE
Review docs

### PR DESCRIPTION
**Pull request by sdkrystian**

### CODE REVIEW

File: include/boost/json/value.hpp [hunk 1]
```diff
@@ -2215,7 +2215,7 @@ class value
 
     /** Return the underlying `std::uint64_t`, or throw an exception.
 
-        If @ref is_int64() is `true`, returns
+        If @ref is_uint64() is `true`, returns
```
Comment (vinniefalco) [inline include/boost/json/value.hpp:2218]: `is_uint64` returns a pointer not a bool, same with the others

Comment (sdkrystian) [reply include/boost/json/value.hpp:2218]: How should it be worded then? "If is_uint64_t() returns a non-null pointer"?

Comment (vinniefalco) [reply include/boost/json/value.hpp:2218]: actually we are going to roll back the pointer change, stay tuned

**Comment by vinniefalco** [comment]
This needs rebasing, since I refactored the `is_array` interfaces

**Comment by vinniefalco** [comment]
This was never rebased and thus cannot be merged.

**Pull request closed without merging**